src/App.tsx

```diff
@@ -1,26 +1,139 @@
+  const [filterBy, setFilterBy] = useState<Filter>(Filter.All);
+  const [errorMessage, setErrorMessage] = useState('');
+  const [newTodoTitle, setNewTodoTitle] = useState('');
+  const [isLoading, setIsLoading] = useState<number[]>([]);
+  const [isDisabledInput, setIsDisabledInput] = useState(false);
+  const [tempTodo, setTempTodo] = useState<Todo | null>(null);
+
+  useEffect(() => {
+    getTodos()
+      .then(setTodos)
+      .catch(() => {
+        setErrorMessage('Unable to load todos');
+        setTimeout(() => setErrorMessage(''), 3000);
+      });
+  }, []);
+
+  const filteredTodos = useMemo(
+    () => getPreparedTodos(todos, filterBy),
+    [todos, filterBy],
+  );
+
   if (!USER_ID) {
     return <UserWarning />;
   }
 
+  function addTodo() {
+    const correctTitle = newTodoTitle.trim();
+
+    const newTempTodo = {
+      id: 0,
+      title: correctTitle,
+      userId: USER_ID,
+      completed: false,
+    };
+
+    setIsDisabledInput(true);
+    setTempTodo(newTempTodo);
+    setIsLoading(ids => [...ids, 0]);
+
+    createTodo({ title: correctTitle, userId: USER_ID, completed: false })
+      .then(newTodo => {
+        setTodos(currentTodos => [...currentTodos, newTodo]);
+        setNewTodoTitle('');
+        setIsDisabledInput(false);
+      })
+      .catch(() => {
+        setErrorMessage('Unable to add a todo');
+        setTimeout(() => setErrorMessage(''), 3000);
+      })
+      .finally(() => {
+        setIsDisabledInput(false);
+        setTempTodo(null);
+        setIsLoading(ids => ids.filter(todoId => todoId !== 0));
+      });
+  }
+
+  function deleteTodoId(todoId: number) {
+    setIsLoading(ids => [...ids, todoId]);
+
+    deleteTodo(todoId)
+      .then(() => {
+        setTodos(currentTodos =>
+          currentTodos.filter(todo => todo.id !== todoId),
+        );
+      })
+      .catch(() => {
+        setErrorMessage('Unable to delete a todo');
+        setTimeout(() => setErrorMessage(''), 3000);
+      })
+      .finally(() => {
+        setIsLoading([]);
+      });
+  }
+
   return (
-    <section className="section container">
-      <p className="title is-4">
-        Copy all you need from the prev task:
-        <br />
-        <a href="https://github.com/mate-academy/react_todo-app-loading-todos#react-todo-app-load-todos">
-          React Todo App - Load Todos
-        </a>
-      </p>
-
-      <p className="subtitle">Styles are already copied</p>
-    </section>
+    <div className="todoapp">
+      <h1 className="todoapp__title">todos</h1>
+
+      <div className="todoapp__content">
+        <Header
+          newTodoTitle={newTodoTitle}
+          setNewTodo={setNewTodoTitle}
+          addTodo={addTodo}
+          onError={setErrorMessage}
+          onDisabled={isDisabledInput}
+        />
+
+        <TodoList
+          todos={filteredTodos}
+          onDelete={deleteTodoId}
+          tempTodo={tempTodo}
+          isLoading={isLoading}
+        />
+
+        {todos.length && (
```

Comment:
This way when you don't have todos it shows 0
![image](https://github.com/mate-academy/react_todo-app-add-and-delete/assets/46566190/37166a27-0547-4226-8670-03762cef47ea)

Just make it boolean
```suggestion
        {!!todos.length && (
```